### ISSUE TITLE
Sometime outputted text overlaps

### ISSUE THREAD

**Issue by VarunAgw** [rprichard/wslbridge#15]
I just installed it today. I have noticed when I paste something or type quickly, the outputted text overlaps over each-other.

Like suppose I paste "a quick brown fox jumps over the lazy dog". It is sometime outputted as "aquic brown fox jumps over the lazy dog" or similar. 

This issue is only limited to stdout. When I execute the code pasted, it works fine, but sometime pasted line appear as gibberish.

**Comment by rprichard** [comment]
I just tried copying-and-pasting that string into wslbridge, and it worked fine for me.

```
$ a quick brown fox jumps over the lazy dog
a: command not found
$ cat
a quick brown fox jumps over the lazy dog
a quick brown fox jumps over the lazy dog
```

**Comment by VarunAgw** [comment]
![quick](https://user-images.githubusercontent.com/4201835/29449122-274df1b0-8417-11e7-9d53-2e43b1412419.png)

**Comment by VarunAgw** [comment]
I think it happens when you quickly paste lot of content. Try it several time.

If you use AutoHotkey, try using this `^v::SendRaw %Clipboard%`. It shows this issue almost everytime with SendRaw API

**Comment by rprichard** [comment]
Based on the screen shot, it looks like you're running wslbridge in a Windows console rather than in a Cygwin/MSYS2 terminal emulator.  (It's easy to tell by looking at the scroll bar.)  It's a little odd to use wslbridge in a console -- mintty would probably work better, or you could run WSL bash without wslbridge.

In any case, I can reproduce the bug.

**Comment by VarunAgw** [comment]
Ah! I actually used windows console to take screenshot. I actually use [ConesoleZ](https://github.com/cbucher/console)

I can reproduce it on both windows console and ConsoleZ. But you are right it works fine with mintty.

**Comment by VarunAgw** [comment]
Are you able to reproduce it now?

**Comment by rprichard** [comment]
Yes, I can reproduce the bug. I think it's a bug in the Cygwin runtime where a concurrent read and write to the tty device interfere with each other. I think I could change wslbridge to use nonblocking I/O, but I'd prefer not to do the work. I have a test case demonstrating the Cygwin bug, so maybe I'll report it.

If you're using a Windows console (or a program based upon one, like ConsoleZ), maybe you could try not using wslbridge? Just run `C:\Windows\System32\bash.exe` (or `/cygdrive/c/Windows/System32/bash.exe` if you're using Cygwin in ConsoleZ). The v2 console in Windows 10 has been extended to support enough of xterm's interface to make WSL work.

**Comment by VarunAgw** [comment]
Unfortunately ConsoleZ have same issues with array keys. That's why I wanted to use wslbridge with it. I guess I am stuck for now.

https://github.com/cbucher/console/issues/335

**Comment by rprichard** [comment]
Oh, yeah, that makes sense. Maybe I'll try to get this issue fixed.
